typedjsonrpc/registry.py

```diff
@@ -126,6 +129,25 @@ def _handle_exceptions(self, method, is_notification=False, msg_id=None):
                 new_error = InternalError.from_error(exc_info, self.json_encoder, debug_url)
                 return Registry._create_error_response(msg_id, new_error)
 
+    def _encode_complete_result(self, result):
+        if isinstance(result, list):
+            return '[' + ','.join([self._encode_single_result(res) for res in result]) + ']'
+        else:
+            return self._encode_single_result(result)
+
+    def _encode_single_result(self, result):
+        msg_id = Registry._get_id_if_known(result)
+        is_notification = msg_id is None
+        encoded = self._handle_exceptions(self.json_encoder.encode,
+                                          is_notification=is_notification,
+                                          msg_id=msg_id,
+                                          params=result)
+        if isinstance(encoded, dict):
```

Comment:
I think this code would make more sense if you made `_handle_exceptions` return a tuple with whether or not there was an error.

That way we wouldn't have to type check!